src/display/annotation_layer.js

```diff
@@ -1671,6 +1671,14 @@ class TextWidgetAnnotationElement extends WidgetAnnotationElement {
               ).valueOf();
               target.step = "";
             } else {
+              // Unfortunately, when the date is "2025-09-23", the parser
+              // converts it to UTC time which may lead to the date being off by
+              // one day depending on the timezone. To workaround this, we
+              // append "T00:00" to the date so that it's parsed as local
+              // time (bug 1989874).
+              if (!value.includes("T")) {
+                value = `${value}T00:00`;
```

Comment:
Would it be possible to add a test case for this?

Reply:
I thought it wasn't possible to set a different timezone for a specific test but it's possible:
https://pptr.dev/api/puppeteer.page.emulatetimezone